src/NServiceBus.TransportTests/When_transaction_level_TransactionScope.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_transaction_level_TransactionScope : NServiceBusTransportTest
+    {
+        [Test]
+        public async Task Should_have_active_transaction()
+        {
+            var messageHandled = new TaskCompletionSource<Tuple<Transaction, Transaction>>();
+
+            await StartPump(
+                context =>
+                {
+                    var currentTransaction = Transaction.Current;
+                    var contextTransaction = context.TransportTransaction.Get<Transaction>();
+                    messageHandled.SetResult(new Tuple<Transaction, Transaction>(currentTransaction, contextTransaction));
+
+                    return Task.FromResult(0);
+                },
+                errorContext => Task.FromResult(ErrorHandleResult.Handled),
+                TransportTransactionMode.TransactionScope);
+            await SendMessage(InputQueueName);
+
+            var transactions = await messageHandled.Task;
+
+            Assert.That(transactions.Item1, Is.Not.Null);
+            Assert.That(transactions.Item2, Is.Not.Null);
+            Assert.AreEqual(transactions.Item1, transactions.Item2);
```

Comment:
good point, I think so yes. There is no guarantee that equal objects remain equal after calling operations (e.g. enlist to the tx) on it. This is most like no issue, but it seems to be more correct indeed.